build-logic/aggregator/src/main/kotlin/org.xtclang.build.aggregator.gradle.kts

```diff
@@ -0,0 +1,61 @@
+import org.gradle.language.base.plugins.LifecycleBasePlugin.ASSEMBLE_TASK_NAME
+import org.gradle.language.base.plugins.LifecycleBasePlugin.BUILD_GROUP
+import org.gradle.language.base.plugins.LifecycleBasePlugin.BUILD_TASK_NAME
+import org.gradle.language.base.plugins.LifecycleBasePlugin.CHECK_TASK_NAME
+import org.gradle.language.base.plugins.LifecycleBasePlugin.CLEAN_TASK_NAME
+
+plugins {
+    base
+}
+
+private class XdkBuildAggregator(project: Project) : Runnable {
+    companion object {
+        private val lifeCycleTasks = listOfNotNull(ASSEMBLE_TASK_NAME, BUILD_TASK_NAME, CHECK_TASK_NAME, CLEAN_TASK_NAME)
+    }
+
+    private val prefix = "[${project.name}]"
+
+    override fun run() {
+        logger.info("$prefix Aggregating included build tasks:")
+        gradle.includedBuilds.forEachIndexed { i, includedBuild ->
+            logger.info("$prefix     Included build #$i: ${includedBuild.name} [project dir: ${includedBuild.projectDir}]")
+        }
+
+        checkStartParameterState()
+        aggregateLifeCycleTasks()
+    }
+
+    private fun aggregateLifeCycleTasks() {
+        lifeCycleTasks.forEach { taskName ->
+            logger.info("$prefix Creating aggregated lifecycle task: ':$taskName' in project '${project.name}'")
+            tasks.named(taskName) {
+                group = BUILD_GROUP
+                description = "Aggregates and executes the '$taskName' task for all included builds."
+                gradle.includedBuilds.forEach { includedBuild ->
+                    dependsOn(includedBuild.task(":$taskName"))
+                    logger.info("$prefix     Attaching: dependsOn(':$name' <- ':${includedBuild.name}:$name')")
+                }
+            }
+        }
+    }
+
+    private fun checkStartParameterState() {
+        val startParameter = gradle.startParameter
+        with (startParameter) {
+            logger.info("""
+            $prefix Start parameter tasks: $taskNames
+            $prefix Start parameter init scripts: $allInitScripts
+        """.trimIndent())
+
+            if (taskNames.count { !it.startsWith("-") && !it.contains("taskTree") } > 1) {
```

Comment:
I'm pretty sure I don't understand the complexity of the issue, but how hard it is to declare that "build" _mustRunAfter_ "clean", so we can use the most commonly command "gradlew clean build". Even better, I'd like to do "gradlew clean build installLocalDist".

Reply:
It deadlocks due to something having to do with the builds still being resolved when starting. It's not a standard or recommended gradle way to run multiple tasks. If you want a talk that does clean build, then write one and call it rebuild. 

mustRunAfter guarantees nothing about task execution, btw.

Reply:
Why would you want to do a "clean build installLocalDist" and not just "installLocalDist"? The dependency resolution should be the same?

Reply:
because I'm doing it about 80 times a day and it would me a bit more annoying